pkg/pod/status.go

```diff
@@ -217,16 +219,28 @@ func createMessageFromResults(results []v1beta1.PipelineResourceResult) (string,
 	return string(bytes), nil
 }
 
-func filterResultsAndResources(results []v1beta1.PipelineResourceResult) ([]v1beta1.TaskRunResult, []v1beta1.PipelineResourceResult, []v1beta1.PipelineResourceResult) {
+func filterResultsAndResources(ctx context.Context, results []v1beta1.PipelineResourceResult) ([]v1beta1.TaskRunResult, []v1beta1.PipelineResourceResult, []v1beta1.PipelineResourceResult) {
 	var taskResults []v1beta1.TaskRunResult
 	var pipelineResourceResults []v1beta1.PipelineResourceResult
 	var filteredResults []v1beta1.PipelineResourceResult
 	for _, r := range results {
 		switch r.ResultType {
 		case v1beta1.TaskRunResultType:
+			cfg := config.FromContextOrDefaults(ctx)
+			aos := v1beta1.ArrayOrString{}
+			if cfg.FeatureFlags.EnableAPIFields == config.AlphaAPIFields {
+				err := aos.UnmarshalJSON([]byte(r.Value))
+				if err != nil {
+					continue
+				}
```

Comment:
this means there is an error, we continue? Just to confirm here: you want to do if you can't unmarshall as array, you want to keep it as string? If so, should it be another else for setting it to be string?

Reply:
Thanks for bring this up!
My original idea is that if we cannot unmarshal the r.Value into ArrayOrString (it can be string, array or object if Chuang's pr got merged), we just continue the for loop and don't throw error. We will add an empty ArrayOrString to the taskRunResult. 

Is it ok if I write it this way? 
```
aos.UnmarshalJSON([]byte(r.Value))
```

Reply:
Please see the other comment. You should definitely explain in details about handling cases about unmarshalling here

Reply:
I am sorry about this I am still catching up ...

Trying to understand why is the error not reported here? 

How are we preventing the users to specify array task results without `alpha` feature flag? Or is this something being addressed somewhere else?

Reply:
Sure np! 
This is an add-on feature of this function. It will include all the cases we support before and also support more cases like unmarshal integer/array of integer into string, and enable us to use `ArrayOrString` for results value

So array results is a new case, it is something we previously don't support.
If user specify array task results without alpha feature flag, this function will be unmarshalled to `array` but it will be converted to string here. So the behaviour won't change.
https://github.com/tektoncd/pipeline/blob/0c008a0c0a6d979f65d281f69ea0c60df3d80169/pkg/entrypoint/entrypointer.go#L208-L211